### PR TITLE
fix: Make attendee info editable only by admin or that user itself

### LINKED ISSUE #7104: Attendee info should be editable by admin or user itself
Like Speaker Information, the attendee info should only be editable by the admin or user itself. Right now, anyone can edit it.

### PR DESCRIPTION
**Pull request by Haider8**
Fixes #7104 

#### Short description of what this resolves:
Right now anybody can edit attendee info which is not correct.

#### Changes proposed in this pull request:
Only admin or that user itself should be able to edit attendee's info.

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by iamareebjamal** [comment]
Please add tests for the logic

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/attendees.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5800316&bid=18988438)

File: app/api/attendees.py [hunk 1]
```diff
@@ -207,6 +207,12 @@ def before_update_object(self, obj, data, kwargs):
         :return:
         """
         order = safe_query_by_id(Order, obj.order_id)
+
+        if not (current_user.is_staff or current_user.id == order.user_id):
+            raise ForbiddenError(
+                {'source': ''}, 'Only admin or that user itself can update attendee info',
```
Comment (iamareebjamal) [inline app/api/attendees.py:213]: {'source': ''} is useless. Provide better pointer or remove it altogether

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/attendees.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5800316&bid=18988438)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/attendees.py  2
         

Clones added
============
- tests/all/integration/api/attendee/test_attendee_api.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5800316&bid=18988438)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/attendees.py  2
         

Clones added
============
- tests/all/integration/api/attendee/test_attendee_api.py  11
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5800316&bid=18988438)

**Comment by Haider8** [comment]
@iamareebjamal Please review

**Pull request merged**